tests/client/test_api.py

```diff
@@ -747,3 +754,39 @@ def test_load_sort(mocked_client):
     ds = api.load(name=dataset, ids=["1str", "2str", "11str"])
     df = ds.to_pandas()
     assert list(df.id) == ["11str", "1str", "2str"]
+
+
+def test_not_aligned_argilla_versions(monkeypatch):
+    from argilla import __version__ as rg_version
+
+    def mock_get_info(*args, **kwargs):
+        return ApiInfo(version="1.0.0")
+
+    def mock_whoami(*args, **kwargs):
+        return User(username="mock")
+
+    monkeypatch.setattr(Status, "get_info", mock_get_info)
+    monkeypatch.setattr(users_api, "whoami", mock_whoami)
+
+    with pytest.warns(
+        UserWarning,
+        match=rf"You're connecting to Argilla Server 1.0.0 using a different client version \({rg_version}\)",
+    ):
+        Argilla()
+
+
+def test_aligned_argilla_versions(monkeypatch):
+    from argilla import __version__ as rg_version
+
+    def mock_get_info(*args, **kwargs):
+        return ApiInfo(version=rg_version)
+
+    def mock_whoami(*args, **kwargs):
+        return User(username="mock")
+
+    monkeypatch.setattr(Status, "get_info", mock_get_info)
+    monkeypatch.setattr(users_api, "whoami", mock_whoami)
+
+    with pytest.warns(None) as record:
```

Comment:
And here I guess there's no need to catch the warning, since no warning will be raised, right?

Reply:
Yes, this test is checking exactly this. The test condition is `len(record) == 0`, which means no warnings are raised.